### PR TITLE
Fix 970 - bad links and sort order in training event display

### PR DESCRIPTION
**Pull request by chrismedrela**

### CODE REVIEW

**Comment by pbanaszkiewicz** [comment]
Ref #970

File: workshops/templates/workshops/all_trainings.html [hunk 1]
```diff
@@ -12,7 +12,7 @@
       </tr>
       {% for training in all_trainings %}
       <tr>
-        <td><a href="{% url 'event_details' training.event %}">{{ training.event__slug }}</a></td>
+        <td><a href="{% url 'event_details' training.event__slug %}">{{ training.event__slug }}</a></td>
```
Comment (pbanaszkiewicz) [inline workshops/templates/workshops/all_trainings.html]: Can we use `.` instead of `__` here?

File: workshops/views.py [hunk 2]
```diff
@@ -1847,7 +1847,7 @@ def all_trainings(request):
         .values('event').annotate(finished=Count('person'))
     finished = {f['event']: f['finished'] for f in finished}
 
-    trainings = Task.objects.filter(role=learner).filter(event__tags=ttt).order_by('event__start') \
+    trainings = Task.objects.filter(role=learner).filter(event__tags=ttt).order_by('-event__start') \
```
Comment (pbanaszkiewicz) [inline workshops/views.py]: Not sure if Greg wanted to change the only sorting way or to have an option to choose sorting. @gvwilson ?

File: workshops/templates/workshops/all_trainings.html (hunk 1 above)
Comment (chrismedrela) [inline workshops/templates/workshops/all_trainings.html]: No, see https://github.com/swcarpentry/amy/blob/develop/workshops/views.py#L1850

File: workshops/views.py (hunk 2 above)
Comment (pbanaszkiewicz) [inline workshops/views.py]: @gvwilson can you please specify what you had in mind [here](https://github.com/swcarpentry/amy/issues/970#issue-173275389)?

Comment (gvwilson) [inline workshops/views.py]: I wanted the default display to be most-recent first.

**Pull request merged**